tic_tac_toe/game.py

```diff
@@ -87,18 +149,52 @@ def move(game, player, position):
     checks before the actual movement is done.
     After registering the movement it must check if the game is over.
     """
-    pass
+    board = game['board']
+    current_player = get_next_turn(game)
+
+    if _board_is_full(board) or game['winner'] != None:
+        raise InvalidMovement('Game is over.')
+
+    if current_player != player:
+        raise InvalidMovement('"' + current_player + '" moves next.')
```

Comment:
Yes, if someone decides to use `'-'` as their token my solution will shatter in a million pieces as well.